packages/gestalt/src/Masonry/Masonry.js

```diff
@@ -321,6 +317,10 @@ export default class Masonry<T> extends React.Component<Props<T>, State> {
     });
   });
 
+  measureContainerAsync = debounce(() => {
```

Comment:
my only gripe here is that we now have `measureContainerAsync` and `measureContainer` and its not clear when to use one vs the other.  

one option is to make `measureContainer` always async and just pass it a callback or make it return a promise.   then we could update componentDidMount to be:
```
this.measureContainer().then(() => {
   //do other things and set some state
});
```
but that's a separate PR for sure 👍